### DISCUSSION TITLE
CLIP skip experiment on style coadapter

### DISCUSSION THREAD

**Discussion by hafriedlander** [TencentARC/T2I-Adapter#50] [Show and tell]
I noticed sometimes when using style images that semantic elements from the style image would affect the result - for instance, a photo with many people would result in people in the output, even if the sketch image and prompt are for something else (i.e. a motorcycle).

I wondered if using earlier hidden layers from the clip image encoding would help.

The answer is "sort of". It's a sometimes useful technique, but also has some drawbacks.

Here's a couple of examples:

Starry night style, final layer results in many stars in the sky, penultimate layer reduces them while still giving a painterly style (but the colors shift too, undesirably)

Style source:

![starry_night](https://user-images.githubusercontent.com/114769/228756160-879bc94f-31dd-4e31-820b-31d8c20d4eba.jpeg)

Final hidden layer:

![style_1st(final)](https://user-images.githubusercontent.com/114769/228756187-b0d8d6b7-a65f-4041-ae47-0a306539e98b.png)

Penultimate layer:

![style_2nd(penultimate)](https://user-images.githubusercontent.com/114769/228756222-5be367ca-cf1e-4fe9-9fd7-bdf5fce5f6be.png)

A random image from unsplash, when using the "Deliberate" model you get a lot of the houses from the style image appearing in the result. Penultimate still gives a similar style without repeating those elements. This is probably my best success so far with this technique:

Style source:

![daniel-j-schwarz-9dnp5vwRUB8-unsplash](https://user-images.githubusercontent.com/114769/228756610-f0e29d38-8efb-4fea-a5d4-352d8bb478d1.jpg)

Final hidden layer:

![styled_1st](https://user-images.githubusercontent.com/114769/228756546-714846ce-f9a5-4360-a787-31454e8b8ec4.png)

Penultimate layer:

![styled_2nd](https://user-images.githubusercontent.com/114769/228756564-a37b1976-4cda-4d38-939c-cfd537e518ad.png)

**Comment by ToTheBeginning** [comment]
Interesting finding, you can also adjust the `Condition weight` of the style to see if it helps. 

I tested your second case in CoAdapter, following are the result when `Condition weight` = 0.9 and prompt = ""
![image](https://user-images.githubusercontent.com/11482921/228774898-12ce7e78-9427-4897-b87e-48b620967572.png)
and when `Condition weight` = 1.0 and prompt = ""
![image](https://user-images.githubusercontent.com/11482921/228776203-b6ea09ae-05cb-4d1f-a0d3-129196540b41.png)
and when `Condition weight` = 0.85 and prompt = "motorcycle"
![image](https://user-images.githubusercontent.com/11482921/228778560-34603cde-2483-4229-8896-ea435d9ea943.png)
